lib/maths/CDataFrameUtils.cc

```diff
@@ -1072,27 +1072,32 @@ CDataFrameUtils::maximizeMinimumRecallForMulticlass(std::size_t numberThreads,
     std::size_t numberSamples{
         static_cast<std::size_t>(std::min(1000.0, rowMask.manhattan()))};
 
-    TStratifiedSamplerPtr sampler;
-    std::tie(sampler, std::ignore) = classifierStratifiedCrossValidationRowSampler(
-        numberThreads, frame, targetColumn, rng, numberSamples, rowMask);
-
-    TSizeVec rowIndices;
-    frame.readRows(1, 0, frame.numberRows(),
-                   [&](TRowItr beginRows, TRowItr endRows) {
-                       for (auto row = beginRows; row != endRows; ++row) {
-                           sampler->sample(*row);
-                       }
-                   },
-                   &rowMask);
-    sampler->finishSampling(rng, rowIndices);
-    std::sort(rowIndices.begin(), rowIndices.end());
     core::CPackedBitVector sampleMask;
-    for (auto row : rowIndices) {
-        sampleMask.extend(false, row - sampleMask.size());
-        sampleMask.extend(true);
+    if (numberSamples < static_cast<std::size_t>(rowMask.manhattan())) {
```

Comment:
Maybe it would be helpfull to add a comment for posterity here why the if-clause is necessary.

Reply:
See [57ba59e](https://github.com/elastic/ml-cpp/pull/1239/commits/57ba59e7c41b8ee444e45b6108b8091c73d2e9fd).